lms/djangoapps/course_home_api/outline/v1/serializers.py

```diff
@@ -21,8 +21,37 @@ def get_url(self, tool):
         return request.build_absolute_uri(url)
 
 
+class DateSummarySerializer(serializers.Serializer):
+    """
+    Serializer for Date Summary Objects.
+    """
+    date = serializers.DateTimeField()
+    date_type = serializers.CharField()
+    description = serializers.CharField()
+    learner_has_access = serializers.SerializerMethodField()
+    link = serializers.SerializerMethodField()
+    title = serializers.CharField()
+    link_text = serializers.CharField()
+    title_html = serializers.CharField()
+
+    def get_learner_has_access(self, block):
+        learner_is_full_access = self.context.get('learner_is_full_access', False)
+        block_is_verified = (getattr(block, 'contains_gated_content', False) or
+                             isinstance(block, VerificationDeadlineDate))
+        return (not block_is_verified) or learner_is_full_access
+
+    def get_link(self, block):
+        if block.link:
+            request = self.context.get('request')
+            return request.build_absolute_uri(block.link)
+        return ''
```

Comment:
So yeah, we will probably want to just import this from the other serializer file, but I'm glad this helped you get here!